lib/project_types/script/ui/printing_spinner.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+require 'shopify_cli'
+
+module Script
+  module UI
+    module PrintingSpinner
+      ##
+      # Creates a single spinner that runs the provided block.
+      # The block can take in a ctx argument that formats printed output to support
+      # printing from within the spin block.
+      #
+      # ==== Attributes
+      #
+      # * +ctx+   - The current context.
+      # * +title+ - Title of the spinner to use
+      #
+      # ==== Options
+      #
+      # * +:auto_debrief+ - Automatically debrief exceptions? Default to true
+      #
+      # ==== Block
+      #
+      # * +ctx+     - Instance of the PrintingSpinnerContext built from the ctx attribute.
+      #             - +ctx.puts(...)+ formats output to enable support for printing within spinners
+      # * +spinner+ - Instance of the spinner. Can call +update_title+ to update the user of changes
+      #
+      def self.spin(ctx, title, auto_debrief: false)
+        StrictSpinner.spin(title, auto_debrief: auto_debrief) do |spinner, *args|
+          Thread.current[:cliui_output_hook] = nil
```

Comment:
Is this a new addition to the CLI::UI? Is that why you recently updated our vendor libraries? If so can you share the changes made so that we might be able to pull this into the core functionality?

Reply:
Nope, I think this was around before I updated the package. The main reason I updated was to grab a race condition fix I made within the spinner implementation